### DISCUSSION TITLE
Version 1.4 Release Date

### DISCUSSION THREAD

**Discussion by AiraYumi** [fltk/fltk#1038] [General]
How long will it be until the 1.4 release?

Is it okay to start updating my application to 1.4 and wayland now?

**Comment by Albrecht-S** [comment]
We're trying hard to release 1.4.0 ASAP but it can still take some days or weeks. There's no release date yet.

> Is it okay to start updating my application to 1.4 and wayland now?

Yes, the API is stable and you can start updating to 1.4 and Wayland. In most cases you can just compile your program with 1.4 and it will just work. Please see also chapter ["Migrating Code from FLTK 1.3 to 1.4"](https://www.fltk.org/doc-1.4/migration_1_4.html) for some details to consider.

If you are using X11 specific code in your app you'll need to take care of this and/or find Wayland replacements for your platform specific code. You can also disable Wayland and thus make your app use X11 (XWayland) on Wayland enabled systems. Please see [README.Wayland.txt](https://github.com/fltk/fltk/blob/master/README.Wayland.txt) for details.

**Comment by Albrecht-S** [comment]
Good news: [FLTK 1.4.0 has been released](https://github.com/fltk/fltk/discussions/1132) on Nov. 17, 2024.

Enjoy!

**Reply by Albrecht-S** [reply]
#### FLTK 1.4.1 has been released on Dec. 12, 2024.

For more info please see the [release announcement](https://github.com/fltk/fltk/discussions/1168).
